app/src/main/java/com/ojasvi/memify/ui/screens/PhotoReasoningScreen.kt

```diff
@@ -0,0 +1,244 @@
+                        null
+                    }
+                } catch (e: Exception) {
+                    null
+                }
+
+                // Call reason function with inputText and bitmap
+                if (bitmap != null) {
+                    viewModel.reason(inputText, bitmap)
+                }
+            }
+        }
+    )
+}
+
+@Composable
+fun PhotoReasoningScreen(
+    uiState: PhotoReasoningUiState = PhotoReasoningUiState.Loading,
+    onReasonClicked: (String, Uri) -> Unit = { _, _ -> }
+) {
+    var userQuestion by rememberSaveable { mutableStateOf("") }
+    var imageUri by rememberSaveable { mutableStateOf<Uri?>(null) }
+
+    val pickMedia = rememberLauncherForActivityResult(
+        ActivityResultContracts.PickVisualMedia()
+    ) { selectedImageUri ->
+        selectedImageUri?.let {
+            imageUri = selectedImageUri
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .padding(all = 16.dp)
+            .verticalScroll(rememberScrollState())
+    ) {
+        AsyncImage(
+            model = imageUri,
+            contentDescription = null,
+            modifier = Modifier
+                .padding(4.dp)
+                .fillMaxWidth(0.5f).align(Alignment.CenterHorizontally)
+        )
+        Card(
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Row(
+                modifier = Modifier.padding(top = 16.dp)
+            ) {
+                IconButton(
+                    onClick = {
+                        pickMedia.launch(
+                            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
+                        )
+                    },
+                    modifier = Modifier
+                        .padding(all = 4.dp)
+                        .align(Alignment.CenterVertically)
+                ) {
+                    Icon(
+                        Icons.Rounded.Add,
+                        contentDescription = stringResource(R.string.add_image),
+                    )
+                }
+                OutlinedTextField(
+                    value = userQuestion,
+                    label = { Text(stringResource(R.string.reason_label)) },
+                    placeholder = { Text(stringResource(R.string.reason_hint)) },
+                    onValueChange = { userQuestion = it },
+                    modifier = Modifier
+                        .fillMaxWidth(0.8f)
+                )
+                TextButton(
+                    onClick = {
+                        if (userQuestion.isNotBlank()) {
+                            imageUri?.let { onReasonClicked(userQuestion, it) }
+                        }
+                    },
+                    modifier = Modifier
+                        .padding(all = 4.dp)
+                        .align(Alignment.CenterVertically)
+                ) {
+                    Text(stringResource(R.string.action_go))
+                }
+            }
```

Comment:
Add clickable to text, no need to use TextButton

Reply:
Done